rtc/hello_rtc/hello_rtc.c

```diff
@@ -32,11 +32,14 @@ int main() {
     rtc_init();
     rtc_set_datetime(&t);
 
+    // clk_sys is >2000x faster than clk_rtc, so datetime is not transferred completely when rtc_get_datetime() is called
+    sleep_us(20);
+
     // Print the time
     while (true) {
         rtc_get_datetime(&t);
         datetime_to_str(datetime_str, sizeof(datetime_buf), &t);
-        printf("\r%s      ", datetime_str);
```

Comment:
why did you remove this? it is presumably there so that garbage isn't left when the datetime str gets shorger

Reply:
There may be garbage, but behind the \0 char. If there is no \0, there is a bug in datetime_to_str. Anyway a few spaces are not the right solution

Reply:
No, if it said 
```
mon 31 january
```
then
```
thu 1 march
```

you don't want it to leave the result (with `\r`)

```
thu 1 marchary
```

Adding spaces is a pertfectly acceptable way of dealing with it in something like this.

Reply:
understood, for me it was strange to overwrite garbage which exists just in a few cases and otherwise adds space garbage.
For this example it may not be of interest => removed this change

Reply:
@Reneg973 this is "overwriting garbage" in the output (i.e. serial terminal) that `printf` is writing to (because `\r` is being used), it's not "overwriting garbage" in the `datetime_str` itself :wink: 
It's kind of similar to doing:
```
printf("\r                                                                          "); // go to start of line, and overwrite it with spaces
printf("\r%s", datetime_str); // go to start of line, and print out datetime_str
```

Reply:
I know, my issue was just that it appends spaces which are not required on most cases